sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/AnalyticsTransmissionTarget.java

```diff
@@ -66,12 +84,61 @@ public synchronized AnalyticsTransmissionTarget getTransmissionTarget(String tra
         /* Reuse instance if a child with the same token has already been created. */
         AnalyticsTransmissionTarget childTarget = mChildrenTargets.get(transmissionTargetToken);
         if (childTarget == null) {
-            childTarget = new AnalyticsTransmissionTarget(transmissionTargetToken);
+            childTarget = new AnalyticsTransmissionTarget(transmissionTargetToken, this);
             mChildrenTargets.put(transmissionTargetToken, childTarget);
         }
         return childTarget;
     }
 
+    public AppCenterFuture<Boolean> isEnabledAsync() {
+        final DefaultAppCenterFuture<Boolean> future = new DefaultAppCenterFuture<>();
+        Analytics.getInstance().postCommand(new Runnable() {
+
+            @Override
+            public void run() {
+                future.complete(isEnabled());
+            }
+        }, future, false);
+        return future;
+    }
+
+    public AppCenterFuture<Void> setEnabledAsync(final boolean enabled) {
+        final DefaultAppCenterFuture<Void> future = new DefaultAppCenterFuture<>();
+        Analytics.getInstance().postCommand(new Runnable() {
+
+            @Override
+            public void run() {
+
+                /*
+                 * Like the relation between AppCenter and Analytics, we cannot change state if one of the parent is disabled.
+                 * If this callback is called then it was already checked that AppCenter and Analytics are both enabled.
+                 */
+                if (areAncestorsEnabled()) {
```

Comment:
We don't have to do any of this if this.isEnabled != enabled (pseudocode).  Also we shouldn't have to update to enabled if Analytics is disabled.

Reply:
I added a unit test that proves otherwise. If you check only 1 parent, this new unit test will fail.